src/main/java/uk/gov/digital/ho/hocs/info/api/TopicService.java

```diff
@@ -20,31 +25,229 @@
+            } else {
+                if (existingTopic.getActive() == true) {
+                    log.debug(
+                            "Unable to create topic, active topic with this name already exists for this parent");
+                } else {
+                    log.debug(
+                            "Unable to create topic, inactive topic with this name already exists for this parent");
+                }
+                throw new ApplicationExceptions.TopicCreationException(
+                        "Topic already exists with this name for the parent topic");
+            }
+        }
+    }
+
+
+    public void deleteParentTopic(UUID parentTopicUUID) {
+
+        log.debug("Deleting parent topic: {}", parentTopicUUID);
+        ParentTopic parentTopic = parentTopicRepository.findByUuid(parentTopicUUID);
+        if (parentTopic == null){
+            log.debug("Unable to delete parent topic, UUID: {} does not exist", parentTopicUUID);
+            throw new ApplicationExceptions.EntityNotFoundException(
+                    "Unable to delete parent topic, UUID: {} does not exist", parentTopicUUID);
+        } else {
+            Set<Topic> childrenTopics = topicRepository.findAllActiveTopicsByParentTopic(parentTopicUUID);
+            if (!childrenTopics.isEmpty()) {
+                for (Topic childTopic : childrenTopics) {
+                    childTopic.setActive(false);
+                    topicRepository.save(childTopic);
+                    log.info("Deleted child topic: {}", childTopic.getDisplayName());
+                    auditClient.deleteParentTopicAudit(parentTopic);
+                }
+            }
+            parentTopic.setActive(false);
+            parentTopicRepository.save(parentTopic);
+            log.info("Deleted parent topic: {}", parentTopic.getDisplayName());
+
+        }
+    }
+
+    public void deleteTopic(UUID topicUUID) {
+
+        log.debug("Deleting topic: {}", topicUUID);
+        Topic topic = topicRepository.findTopicByUUID(topicUUID);
+        if (topic == null){
+            log.debug("Unable to delete  topic, UUID: {} does not exist", topicUUID);
+            throw new ApplicationExceptions.EntityNotFoundException
+                    ("Unable to delete topic, UUID: {} does not exist", topicUUID);
+        } else {
+            topic.setActive(false);
+            topicRepository.save(topic);
+            log.info("Deleted topic: {}", topic.getDisplayName());
+            auditClient.deleteTopicAudit(topic);
+        }
+    }
+
+    public void reactivateParentTopic(UUID parentTopicUUID) {
+
+        log.debug("Reactivating parent topic: {}", parentTopicUUID);
+        ParentTopic parentTopic = parentTopicRepository.findByUuid(parentTopicUUID);
+        if (parentTopic == null){
+            throw new ApplicationExceptions.EntityNotFoundException("Parent topic with UUID {} does not exist", parentTopicUUID.toString());
+        }
+        if (parentTopic.getActive() == true) {
```

Comment:
Probably don't need this If, it's an idempotent action, if it's already active then making it active again is OK, it will of course create an additional audit log line but that's not a problem